src/main/java/org/recommendation/service/MovieRecommenderSystem.java

```diff
@@ -0,0 +1,108 @@
+package org.recommendation.service;
+
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import java.io.BufferedReader;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Scanner;
+
+import org.recommendation.data.DataInitializer;
+import org.recommendation.data.MovieDataInitializer;
+import org.recommendation.data.UserDataInitilizer;
+import org.recommendation.data.GenreDataInitializer;
+import org.recommendation.log.Logger;
+import org.recommendation.log.SoutLogger;
+
+public class MovieRecommenderSystem implements RecommenderSystem {
+    // using singleton design pattern
+    private static MovieRecommenderSystem recommenderSystem;
+
+    private MovieRecommenderSystem() {
+    }
+
+    private static final String MOVIEFILEPATH = "src/main/java/org/recommendation/data/raw/movie.csv";
+    private static final String GENREFILEPATH = "src/main/java/org/recommendation/data/raw/genre.csv";
+    private static final String RATINGFILEPATH = "src/main/java/org/recommendation/data/raw/ratings.csv";
+
+
+    public static MovieRecommenderSystem getRecommenderSystem() {
+        if (Objects.isNull(recommenderSystem)) {
+            recommenderSystem = new MovieRecommenderSystem();
+        }
+        return recommenderSystem;
+    }
+
+    public void printRecommendations(final String userInput) {
+        Logger logger = new SoutLogger();
+        logger.info("Hold on for a while, starting recommendation process for your inputs");
+        Scanner scanner = new Scanner(System.in);
+        DataInitializer dataInitializer;
+        dataInitializer = new GenreDataInitializer();
+        initializer(dataInitializer, GENREFILEPATH);
+        dataInitializer = new MovieDataInitializer();
+        initializer(dataInitializer, MOVIEFILEPATH);
+        dataInitializer = new UserDataInitilizer();
+        initializer(dataInitializer, RATINGFILEPATH);
+        String result = null;
+        String input = userInput;
+        TopItemFinder topItemFinder;
+        switch (userInput) {
```

Comment:
can we replace this switch case to any design patterm?

Reply:
Replaced with Command Design Pattern